### PR TITLE
Bugfix/change select value to match contact type scope

### LINKED ISSUE #1138: In Case Contact Report, Contact Types filter do not when one or more options are selected
Sample params of a report request:

{
"start_date"=>"2020-05-19", 
"end_date"=>"2020-10-20", 
"supervisor_ids"=>["", "2"], 
"creator_ids"=>["", "1"], 
"contact_type_ids"=>["", "1", "2"], 
"contact_type_group_ids"=>["", "1", "2", "3", "4", "5", "6"], 
"want_driving_reimbursement"=>"true", 
"contact_made"=>"false", 
"has_transitioned"=>"", 
"commit"=>"Download Report"
}

Identified a bunch of other bugs relating to scopes of CaseContact used in CaseContactReport filters at:
- /case_contact_reports

3. Contact Types filters **do not** work properly. If none is selected, it will accept all `contact_types`, which is expected. If one or more options are selected, the query is comparing `contact_types.name` against `contact_type.ids`.

```rb
# app/models/case_contact.rb
  scope :contact_type, ->(contact_type = nil) {
    if contact_type.present?
      joins(:contact_types)
        .where("contact_types.name in (?)", contact_type)
    end
  }
```

There are 2 ways to fix this:

1. Change to use `contact_types`.`id` for filtering. It is similar to the bug of this issue:
```rb
  scope :contact_type, ->(contact_type_ids = nil) {
    if contact_type.present?
      joins(:contact_types)
        .where(contact_types.id: contact_type_ids)
    end
  }
```

2. Change the generate values of options element of Contact Type to `name` from `id`:

image
image

_Originally posted by @erik-trantt in https://github.com/rubyforgood/casa/issues/1117#issuecomment-712247123_

### PR DESCRIPTION
**Pull request by cassianoblonski**
### What github issue is this PR for, if any?
Resolves #1138 

### What changed, and why?
The scope on `models/case_contact.rb` to match the design on the Select, which is using ids in the value field, previously the scope was expecting the name of ContactType, not the ID.

### How is this tested? (please write tests!) 💖💪
Run the entire Rspec Test Suite and I made a spec to cover this scope.

### CODE REVIEW

**Comment by jmkoni** [comment]
Looks good @cassianoblonski! just need to run `standardrb --fix`

**Review by jmkoni — APPROVED** [review]

**Comment by cassianoblonski** [comment]
> Looks good @cassianoblonski! just need to run `standardrb --fix`

done!

**Review by compwron — APPROVED** [review]

**Comment by compwron** [comment]
Looks like some tests are failing - please fix :)

**Comment by cassianoblonski** [comment]
@compwron hey take a look at the build end its failing but I don't recognize this error 

image

**Comment by compwron** [comment]
@cassianoblonski That's a warning, not an error. The error is from standardrb https://travis-ci.org/github/rubyforgood/casa/builds/738835239
Screen Shot 2020-10-25 at 6 16 48 PM

**Comment by cassianoblonski** [comment]
> @cassianoblonski That's a warning, not an error. The error is from standardrb https://travis-ci.org/github/rubyforgood/casa/builds/738835239
> Screen Shot 2020-10-25 at 6 16 48 PM

ohh, fixing it now, thanks for the fast reply

File: spec/models/case_contact_spec.rb [hunk 1]
```diff
@@ -99,4 +99,24 @@
...
+        youth_type = create(:contact_type, name: "Youth", contact_type_group: group)
+        supervisor_type = create(:contact_type, name: "Supervisor", contact_type_group: group)
+        parent_type = create(:contact_type, name: "Parent", contact_type_group: group)
+
+        case_contacts_to_match = [
+          create(:case_contact, contact_types: [youth_type, supervisor_type]),
+          create(:case_contact, contact_types: [supervisor_type]),
+          create(:case_contact, contact_types: [youth_type, parent_type])
+        ]
+        case_contact_to_not_match = create(:case_contact, contact_types: [parent_type])
```
Comment (jmkoni) [inline spec/models/case_contact_spec.rb:116]: To fix this, you should just need to do this:
```
create(:case_contact, contact_types: [parent_type])
```

**Comment by jmkoni** [comment]
@cassianoblonski There is still a linting error. 
```
  spec/models/case_contact_spec.rb:116:9: Lint/UselessAssignment: Useless assignment to variable - `case_contact_to_not_match`. Did you mean `case_contacts_to_match`?
823
824
```
make sure it exits successfully (no text after running the linter).

**Comment by cassianoblonski** [comment]
> @cassianoblonski There is still a linting error.
> 
> ```
>   spec/models/case_contact_spec.rb:116:9: Lint/UselessAssignment: Useless assignment to variable - `case_contact_to_not_match`. Did you mean `case_contacts_to_match`?
> 823
> 824
> ```
> 
> make sure it exits successfully (no text after running the linter).

I think its all ok now, thanks for the replies

**Comment by jmkoni** [comment]
@cassianoblonski thank you for all your work!!! 🎉

**Pull request merged**